src/sql/src/func.rs

```diff
@@ -1705,7 +1705,7 @@ macro_rules! privilege_fn {
                             LEFT JOIN mz_roles ON
                                     mz_internal.mz_aclitem_grantee(privilege) = mz_roles.id
                         WHERE
-                            mz_roles.oid = $1
+                            mz_internal.mz_aclitem_grantee(privilege) = 'p' OR pg_has_role($1, mz_roles.oid, 'USAGE')
```

Comment:
What does the `mz_internal.mz_aclitem_grantee(privilege) = 'p'` do? That wasn't noted in the linked issue.